### PR TITLE
fix: support for windows file paths

### PR DESCRIPTION
**Pull request by marbemac**
BREAKING CHANGE: authority is now baseUri, and is expected to be passed into the `resolve` function as a string

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,18 +50,19 @@
...
     "lodash": "4.x.x",
-    "urijs": "1.x.x"
+    "urijs": "1.x.x",
+    "vscode-uri": "2.x.x"
   },
   "devDependencies": {
-    "@stoplight/scripts": "7.0.0",
-    "@types/jest": "^24.0.13",
-    "@types/lodash": "4.14.133",
+    "@stoplight/scripts": "5.x.x",
```
Comment (wmhilton) [inline package.json]: bahaha you had to _downgrade_ `@stoplight/scripts`? Uh oh!

File: src/__tests__/resolver.spec.ts [hunk 2]
```diff
@@ -115,6 +115,74 @@ describe('resolver', () => {
...
+            b_name: 'b',
+            b_inner: {
+              $ref: './models/c.json',
+            },
+          },
+        },
+        'c:/models/c.json': {
+          c_name: 'c',
+          network: {
+            $ref: 'D:\\network.json#/inner',
```
Comment (wmhilton) [inline src/__tests__/resolver.spec.ts]: 👀 cooooool

File: src/runner.ts [hunk 3]
```diff
@@ -241,19 +250,23 @@ export class ResolveRunner implements Types.IResolveRunner {
...
+    if (ref.toString().charAt(0) !== '#') {
+      const isFile = this.isFile(ref);
 
       // if we're working with a file, resolve any path diferences and make sure the scheme is set
       if (isFile) {
-        ref = ref.absoluteTo(this.authority).scheme('file');
+        let absRef = ref.toString();
+        if (!ref.is('absolute')) {
+          if (this.authority.toString()) {
+            absRef = join(dirname(this.authority.toString()), absRef);
```
Comment (wmhilton) [inline src/runner.ts]: I'm kinda hesitant to use `path.join` and `path.dirname` on URIs - in this case it works because they are relative and we know it's a file... but I'm wonder if there's an edge case we might not be aware of.

**Comment by wmhilton** [comment]
Looking good so far! I think `baseUri` as a string (instead of `authority` as a URI object) is a good idea too.

File: package.json (hunk 1 above)
Comment (P0lip) [reply package.json]: Yeah, Rollup is a bit of an overkill for cases like this one.

File: src/runner.ts [hunk 4]
```diff
@@ -420,4 +441,29 @@ export class ResolveRunner implements Types.IResolveRunner {
...
       .toString();
   }
+
+  private isFile(ref: uri.URI): boolean {
+    const scheme = ref.scheme();
+
+    if (scheme === 'file') return true;
+
+    if (!scheme) {
+      // if no scheme set, and ref starts with a '/', assume it's a file
```
Comment (P0lip) [inline src/runner.ts]: What if it starts with a drive letter, i.e. `C:\\`?

Comment (marbemac) [reply src/runner.ts]: Covered - https://github.com/stoplightio/json-ref-resolver/pull/57/files#diff-02f2832865abf89619d47f8c83164733R137

File: src/runner.ts (hunk 3 above)
Comment (marbemac) [reply src/runner.ts]: Hmm, I can't think of any - I added some test cases to try to cover the edge cases I could think of though.

Comment (wmhilton) [reply src/runner.ts]: I posted an alternative to `dirname` in Slack. I'm actually helping paint the house this weekend, but... these are the things I end up thinking about. :)

Comment (wmhilton) [reply src/runner.ts]: Or... shouldn't we just use URI.js here to compute the absolute path?

Comment (P0lip) [reply src/runner.ts]: Does not work under Windows. See Studio branch and the comment I left.
